### DISCUSSION TITLE
How are you planning the testing part

### DISCUSSION THREAD

**Discussion by FareesHussain** [google/modernstorage#14] [Q&A]
So far I can guess that we can't add Unit tests for the media store library as we require `context` or `StorageLocation` for most of the functions.

I think we need to add Instrumentation tests in the sample app to test the functions of the android library correct me (not sure)
If we have to add Instrumentation tests for the library do we have to add mock activities.

In case we add Instrumentation tests are we expecting `UI Automator` here, as using `Espresso` we can't have complete access to the app and granting storage permissions.

Please let me know, I'm interested to work on any of these.

**Comment by yrezgui** [comment]
Hi @FareesHussain, I wasn't comfortable sharing more about the testing strategy as it was unclear where our experiment back then would lead. As I wrote in this [announcement](https://github.com/google/modernstorage/discussions/68), I'm more comfortable now regarding what we want to do with the library. The testing strategy is to focus on instrumentation tests as I've seen a lot of unexpected behaviours regarding the storage implementation on previous Android releases (Both platform & OEM specific differences).

I'm looking at Firebase Testlab to check on physical devices but happy to discuss more with you.

**Reply by FareesHussain** [reply]
@yrezgui Can't we do Espresso or UiAutomator instead?

**Reply by yrezgui** [reply]
My bad, I wrote too quickly. First UiAutomator (we already have some tests but not fully working before API 24 I think) and after that, we'll have more tests, including for the sample app. If you want to contribute, I'll be glad to review any PR!

**Reply by FareesHussain** [reply]
Sure I'd like to

Is there any tests pending that I can work on?

**Reply by yrezgui** [reply]
Coverage isn't great currently, I should write a todo list of them. Do you have any ideas currently?

**Reply by FareesHussain** [reply]
In that case I'll check what I can do till you make a todo list of the tests
